docs/federate.md

```diff
@@ -0,0 +1,135 @@
+Setting up Federation
+=====================
+
+Federation is the process by which users on different servers can participate
+in the same room. For this to work, those other servers must be able to contact
+yours to send messages.
+
+The ``server_name`` configured in the Synapse configuration file (often
+``homeserver.yaml``) defines how resources (users, rooms,...) will be
+identified (ie: ``@user:example.com``, ``#room:example.com``). By
+default, it is also the domain that other servers will use to
+try to reach your server via port 8448. This is easy to set
+up and will work provided you set the ``server_name`` to match your
+machine's public DNS hostname, and provide Synapse with a TLS certificate
+which is valid for your ``server_name``.
+
+Once you have completed the steps necessary to federate, you should be able to
+ [connect to your server with a client](../README.rst#registering-a-new-user-from-a-client).
+and then join a room via federation. (A good place to start is ``#synapse:matrix.org``
+ - a room for Synapse admins.)
+
+
+## Delegation
+
+For a more flexible configuration, you can have ``server_name``
+resources (ie: ``@user:example.com``) served by a different host and
+port (ie: ``synapse.example.com:443``). There are two ways to do this:
+
+- adding a ``/.well-known/matrix/server`` URL served on ``https://example.com``.
+- adding a DNS ``SRV`` record in the DNS zone of domain
+  ``example.com``.
+
+For both methods let's say you want to run your server at
+``synapse.example.com`` on port ``443`` (instead of ``8448``), but you
+want to have your Matrix user-ids look like ``@user:example.com``.
+
+Without configuring delegation, the matrix federation will
+expect to find your resources via ``example.com:8448``. The
+following methods allow you to provide a different server and port for
+``*:example.com`` resources.
+
+### .well-known delegation
+
+To use this method, you need to be able to alter the
+``server_name`` 's https server to serve the ``/.well-known/matrix/server``
+URL. Having an active server (with a valid ``SSL`` certificate) serving your
+``server_name`` domain is out of the scope of this documentation.
+
+The URL ``https://<server_name>/.well-known/matrix/server`` should
+return a JSON structure containing the key ``m.server`` like so:
+
+    {
+	    "m.server": "<synapse.server.name>:<yourport>"
+    }
+
+In our example, this would mean that URL ``https://example.com/.well-known/matrix/server``
+should return:
+
+    {
+	    "m.server": "synapse.example.com:443"
+    }
+
+Note, specifying a port is optional and will default to 443.
```

Comment:
no, it won't.

If you don't specify a port, it will go on to do a SRV lookup, which may in turn specify a port. If there is no SRV record on the target of the delegation, it will default to port 8448.

Reply:
Reworded